src/cpu/flexcpu/inflight_inst.cc

```diff
@@ -52,11 +53,62 @@ InflightInst::InflightInst(ThreadContext* backing_context,
     _pcState(pc_)
 {
     _timingRecord.creationTick = curTick();
+    // Tick's are unsigned, so effectively, -1 is MAX_INT
+    _timingRecord.decodeTick = -1;
+    _timingRecord.issueTick = -1;
+    _timingRecord.beginExecuteTick = -1;
+    _timingRecord.effAddredTick = -1;
+    _timingRecord.beginMemoryTick = -1;
+    _timingRecord.completionTick = -1;
+    _timingRecord.commitTick = -1;
+    _timingRecord.squashTick = -1;
     staticInst(inst_ref);
 }
 
 InflightInst::~InflightInst()
 {
+#if TRACING_ON
+    if (DTRACE(FlexPipeView))
+    {
+        if (!static_cast<bool>(instRef))
+        {
+            DPRINTFR(FlexPipeView,
+            "pipe;%llu;%llx;%llx;null;\
+%lld;%lld;%lld;%lld;%lld;%lld;%lld;%lld;%lld\n",
+                     this->issueSeqNum(),
+                     this->_pcState.npc(),
+                     this->_pcState.nupc(),
+                     (long long)(this-_timingRecord.creationTick),
```

Comment:
Is the cast required? Or should you use `%llu` in the format string?

Reply:
Initially, I used `%llu`, but the output for `MAX_INT` would be 20 characters long, i.e. `18446744073709551616`, instead of `-1`, which is 2 characters long if I use `%lld`. This is an optimization for the debug output size. I don't know whether it is worth it, but there are lots of places that the tick is `-1`. Also, I remember that if I didn't cast the tick to `long long`, it will output `2^32 - 1` instead of `-1`. I'll investigate this again.

Reply:
See comment above. I would output the huge number to the tracefile, but then render it better in the viewer.